diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/utils/ASTNodeUtilsTest.kt

```diff
@@ -61,6 +75,439 @@ class ASTNodeUtilsTest {
+    }
+
+    @Test
+    fun `test getAllChildrenWithType`(){
+        val code = """
+            class Test() {
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        applyToCode(code){node ->
+            val list = mutableListOf<String>()
+            if (node.getAllChildrenWithType(CLASS).isNotEmpty()) {
+                node.getAllChildrenWithType(CLASS).forEach {
+                    list.add(it.text)
+                }
+                Assert.assertEquals(list, listOf(code))
+            }
+            if (node.getAllChildrenWithType(IDENTIFIER).isNotEmpty()) {
+                Assert.assertEquals(node.getAllChildrenWithType(IDENTIFIER)[0].text, node.getAllIdentifierChildren()[0].text)
+                Assert.assertEquals(node.getAllChildrenWithType(IDENTIFIER)[0].text, node.getIdentifierName()!!.text)
+            }
+        }
+    }
+
+    @Test
+    fun `test getFirstChildWithType`(){
+        val code = """
+            class Test() {
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        applyToCode(code){node ->
+            if(node.getAllChildrenWithType(IDENTIFIER).isNotEmpty()){
+                Assert.assertEquals(node.getFirstChildWithType(IDENTIFIER)!!.text, node.getAllChildrenWithType(IDENTIFIER)[0].text)
+                Assert.assertEquals(node.getFirstChildWithType(IDENTIFIER)!!.text, node.getIdentifierName()!!.text)
+            }
+        }
+    }
+
+    @Test
+    fun `test hasChildOfType`(){
+        val code = """
+            class Test {
+                val x = 0
+            }
+        """.trimIndent()
+        applyToCode(code){node ->
+            if (node.getIdentifierName() != null)
+                Assert.assertTrue(node.hasChildOfType(IDENTIFIER))
+        }
+    }
+
+    @Test
+    fun `test hasAnyChildOfTypes`(){
+        val code = """
+            class Test {
+                val x = 0
+            }
+        """.trimIndent()
+        applyToCode(code){node ->
+            if (node.getAllChildrenWithType(IDENTIFIER).isNotEmpty() || node.getAllChildrenWithType(CLASS).isNotEmpty()) {
+                val param = arrayOf(IDENTIFIER, CLASS)
+                Assert.assertTrue(node.hasAnyChildOfTypes(*param))
+            }
+        }
+    }
+
+    @Test
+    fun `test findChildBefore`(){
+        val code = """
+            class Test() {
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        applyToCode(code){node ->
+            if (node.findChildBefore(CLASS_BODY, CLASS) != null)
```

Comment:
Does this condition work? IIRC, CLASS and CLASS_BODY are usually on different levels in AST.